src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -1947,6 +1947,25 @@ def unregister_provider(cmd, resource_provider_namespace, wait=False):
     _update_provider(cmd.cli_ctx, resource_provider_namespace, registering=False, wait=wait)
 
 
+def register_management_group(cmd, resource_provider_namespace, management_group_id, wait=False):
+    import time
+    rcf = _resource_client_factory(cmd.cli_ctx)
+    r = rcf.providers.register_at_management_group_scope(resource_provider_namespace, management_group_id)
+    if r.registration_state == 'Registered':
+        return
+
+    if wait:
+        while True:
+            time.sleep(10)
+            rp_info = rcf.providers.get(resource_provider_namespace)
+            if rp_info.registration_state == 'Registered':
+                break
```

Comment:
How about declaring `register_management_group` as a long running operation in swagger in the future? 
Can the logic of judging whether the registration has been completed be replaced by the judgment of `ARMPolling` according to the status code?